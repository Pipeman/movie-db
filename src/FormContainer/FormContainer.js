import React, { Component } from "react";
import MovieInput from "../MovieInput/MovieInput";

class FormContainer extends Component {

  constructor() {
    super();

    this.state = {
      searchedMovie: "",
    }
    
    this.setMovieTitleHandler = this.setMovieTitleHandler.bind(this);
  }

  setMovieTitleHandler(event) {
    this.setState({
      "searchedMovie": event.target.value,
    });

    console.log(event.target.value)
  }

  render() {
    return (
      <form id="movie-search">
        <MovieInput
          value={this.state.searchedMovie}
          handleChange={this.setMovieTitleHandler}
        />
      </form>
    );
  }
}

export default FormContainer;