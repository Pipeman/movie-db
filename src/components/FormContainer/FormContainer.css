.cards-container {
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  grid-gap: 2rem 2rem;
  justify-items: center;
  padding: 2rem 1rem;
}

@media screen and (min-width: 460px) {
  .cards-container {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media screen and (min-width: 660px) {
  .cards-container {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media screen and (min-width: 860px) {
  .cards-container {
    grid-template-columns: repeat(4, 1fr);
    padding: 2rem 2rem;
  }
}