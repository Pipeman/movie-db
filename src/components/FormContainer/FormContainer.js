import React, { Component } from "react";
import MovieInput from "../MovieInput/MovieInput";
import FilmCard from "../FilmCard/FilmCard";
import { searchMovie } from "../../services/movieDB.service";

class FormContainer extends Component {

  constructor() {
    super();

    this.state = {
      searchedMovie: "",
      foundMovies: [],
    }
    
    this.setSearchedMovieHandler = this.setSearchedMovieHandler.bind(this);
  }

  setSearchedMovieHandler(event) {
    this.setState({
      "searchedMovie": event.target.value,
    });
    
    if (event.target.value !== "") {
      this.fetchSearchedMovies(event.target.value);
    }
  }

  async fetchSearchedMovies(searchedMovie) {
    const foundMovies = await searchMovie(searchedMovie);
    this.setState({
      foundMovies: foundMovies.results,
    });
  }

  render() {
    return (
      <form>
        <MovieInput
          handleChange={this.setSearchedMovieHandler}
        />
        {
          this.state.foundMovies.map(film => <FilmCard info={film} key={film.id} />)
        }
      </form>
    );
  }
}

export default FormContainer;