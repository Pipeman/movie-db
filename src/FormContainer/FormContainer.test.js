import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import FormContainer from "./FormContainer";

let wrapper;

beforeEach(() => {
  wrapper = mount(
    <FormContainer/>
  );
});

it("shoudld render the component correctly", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("should handle the change in the input", () => {
  const value = "cane";
  wrapper.find("input").simulate("change", {
    target: {
      value,
    }
  });
  expect(wrapper.find("input").props().placeholder).toBe(value);
  expect(toJson(wrapper)).toMatchSnapshot();
})