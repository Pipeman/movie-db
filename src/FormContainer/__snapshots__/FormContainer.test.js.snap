// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoudld render the component correctly 1`] = `
<FormContainer>
  <form
    id="movie-search"
  >
    <MovieInput
      handleChange={[Function]}
      value=""
    >
      <div
        className="movie-input-copntainer"
      >
        <label
          htmlFor="movie-name"
        >
          Movie to search
        </label>
        <input
          className="movie-input"
          id="movie-name"
          onChange={[Function]}
          placeholder=""
          type="text"
        />
      </div>
    </MovieInput>
  </form>
</FormContainer>
`;

exports[`should handle the change in the input 1`] = `
<FormContainer>
  <form
    id="movie-search"
  >
    <MovieInput
      handleChange={[Function]}
      value="cane"
    >
      <div
        className="movie-input-copntainer"
      >
        <label
          htmlFor="movie-name"
        >
          Movie to search
        </label>
        <input
          className="movie-input"
          id="movie-name"
          onChange={[Function]}
          placeholder="cane"
          type="text"
        />
      </div>
    </MovieInput>
  </form>
</FormContainer>
`;
